src/widgets.jl

```diff
@@ -328,6 +328,59 @@ end
 button(label::Union{String,Symbol}; widget=nothing, observable=nothing, own=nothing) =
     button(; label=label, widget=widget, observable=observable, own=own)
 
+######################### ColorButton ###########################
+
+struct ColorButton{C} <: InputWidget{Nothing}
+    observable::Observable{C}
+    widget::GtkColorButtonLeaf
+
+    function ColorButton{C}(observable::Observable{C}, widget) where C <: Colorant
+        obj = new(observable, widget)
+        gc_preserve(widget, obj)
+        obj
+    end
+end
+
+colorbutton(observable::Observable{C}, widget::GtkColorButtonLeaf) where C <: Colorant =
+    ColorButton{C}(observable, widget)
+
+"""
+    colorbutton(color; widget=nothing, observable=nothing)
+    colorbutton(; color=nothing, widget=nothing, observable=nothing)
+
+Create a push button with color `color`. Clicking opens a color picker. Optionally provide:
+  - a GtkButton `widget` (by default, creates a new one)
+  - the (Observables.jl) `observable` coupled to this button (by default, creates a new observable)
+"""
+function colorbutton(;
+                color::C = RGBA(0, 0, 0, 0),
+                widget=nothing,
+                observable=nothing,
+                own=nothing) where C <: Colorant
+    obsin = observable
+    if observable === nothing
+        observable = Observable(color)
+    end
+    if own === nothing
+        own = observable != obsin
+    end
+    gcolor = Gtk.GdkRGBA(color.r, color.g, color.b, color.alpha)
+    if widget === nothing
+        widget = GtkColorButton(gcolor)
+    else
+        set_gtk_property!(widget, :rgba, gcolor)
+    end
+
+    id = signal_connect(widget, "color-set") do w
+        color = get_gtk_property(widget, :rgba, Gtk.GdkRGBA)
+        setindex!(observable, RGBA(color.r, color.g, color.b, color.a))
+    end
```

Comment:
One thing that's missing is the opposite direction of control:
```julia
mycolor = Observable(RGBA(1, 0, 0))
btn = colorbutton(observable = mycolor)
mycolor[] = RGBA(0, 1, 0)    # this should update the graphical display, but currently it doesn't
```
That's what the whole `init_observable2widget` is about, and why you need `own` and `preserve`.